src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp.Client/PersistedAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System.Security.Claims;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+
+namespace BlazorWeb_CSharp.Client;
+
+public class PersistedAuthenticationStateProvider(PersistentComponentState persistentState) : AuthenticationStateProvider
+{
+    private static readonly Task<AuthenticationState> _unauthenticatedTask =
+        Task.FromResult(new AuthenticationState(new ClaimsPrincipal(new ClaimsIdentity())));
+
+    public override Task<AuthenticationState> GetAuthenticationStateAsync()
+    {
+        // REVIEW: Is TryTakeFromJson correctly annotated? "|| userInfo is null" should not be necessary.
+        if (!persistentState.TryTakeFromJson<UserInfo>(nameof(UserInfo), out var userInfo) || userInfo is null)
```

Comment:
I can just update the nullability of `TryTakeFromJson` as part of this PR, right? I think it should be just `[MaybeNullWhen(false)] out TValue instance` rather than `out TValue? instance`.

https://github.com/dotnet/aspnetcore/blob/34e3b750e5357b057cc92fd9250c7929b1b6e1d2/src/Components/Components/src/PersistentComponentState.cs#L88

That seems to be what [everyone else does.](https://grep.app/search?q=%5BMaybeNullWhen%28false%29%5D%20out%20TValue)

Reply:
I want to make the nullability change to `TryTakeFromJson`, but don't want to make any API changes this late in RC2 no matter how small. We'll live with the redundant `|| userInfo is null` check for now.